tau.h

```diff
@@ -85,6 +115,34 @@ class TauCorrProvider : public CorrectionsBase<TauCorrProvider> {
         return final_p4;
     }
 
+    float getSF(const LorentzVectorM& Tau_p4, int Tau_decayMode, int Tau_genMatch, const std::string& wpVSe, const std::string& wpVSmu, const std::string& wpVSjet, UncSource source, UncScale scale, const std::string& genuineTau_SFtype) const
+    {
+        if(isTwoProngDM(Tau_decayMode)) throw std::runtime_error("no SF for two prong tau decay modes");
+        const GenLeptonMatch genMatch = static_cast<GenLeptonMatch>(Tau_genMatch);
+        if(genMatch == GenLeptonMatch::Tau) {
+            const UncScale tau_had_scale = sourceApplies(source, Tau_p4, Tau_decayMode, genMatch)
+                                           ? scale : UncScale::Central;
+            const std::string& scale_str = getScaleStr(tau_had_scale);
+            if(wpVSe < 3 || wpVSmu < 4){
```

Comment:
- define WP enums instead directly using 3&4
- only enter inside if when tau_had_scale != central
- evaluate for non-central is common -> avoid code duplications
- for high pt taus additional unc is bigger than 5%

Reply:
ok